src/libstd/net/ip.rs

```diff
@@ -1488,20 +1495,24 @@ impl Ipv6Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Display for Ipv6Addr {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Long enough for the longest possible Ipv6: 45 bytes
+        let mut buf = [0u8;48];
```

Comment:
Perhaps put the length in a `const` so you can reuse it while calculating `len`. Maybe name it `MAX_IPV6_LEN` or something?